definitions/npm/history_v4.9.x/flow_v0.25.x-/history_v4.9.x.js

```diff
@@ -92,32 +96,44 @@ declare module "history" {
   };
 
   declare interface IHashHistory {
-    length: number,
-    location: HashLocation,
-    action: Action,
-    push(path: string, state?: {}): void,
-    push(location: $Shape<HashLocation>): void,
-    replace(path: string, state?: {}): void,
-    replace(location: $Shape<HashLocation>): void,
-    go(n: number): void,
-    goBack(): void,
-    goForward(): void,
-    listen((location: HashLocation, action: Action) => void): void,
-    block(message: string): typeof Unblock,
-    block((location: HashLocation, action: Action) => string): typeof Unblock,
-    push(path: string): void,
+    length: number;
+    location: HashLocation;
+    action: Action;
+    push(path: string, state?: {}): void;
+    push(location: $Shape<HashLocation>): void;
+    replace(path: string, state?: {}): void;
+    replace(location: $Shape<HashLocation>): void;
+    go(n: number): void;
+    goBack(): void;
+    goForward(): void;
+    listen((location: HashLocation, action: Action) => void): void;
+    block(message: string): typeof Unblock;
+    block((location: HashLocation, action: Action) => string): typeof Unblock;
+    push(path: string): void;
   }
 
   declare export type HashHistory = IHashHistory;
 
   declare type HashHistoryOpts = {
     basename?: string,
-    hashType: "slash" | "noslash" | "hashbang",
+    hashType: 'slash' | 'noslash' | 'hashbang',
     getUserConfirmation?: (
       message: string,
-      callback: (willContinue: boolean) => void,
+      callback: (willContinue: boolean) => void
     ) => void,
   };
 
   declare function createHashHistory(opts?: HashHistoryOpts): HashHistory;
+
+  declare type LocationType = {
+    pathname?: string,
+    search?: string,
+    hash?: string,
+  };
```

Comment:
There's already `BrowserLocation`, `MemoryLocation`, `HashLocation` (which should be discussed BTW...)   
Why creating another one?

Reply:
Yeah I see what you mean here. I can remove this type and use a union of all Location types in the following 2 functions